contracts/JBETHPaymentTerminal.sol

```diff
@@ -246,7 +247,12 @@ contract JBETHPaymentTerminal is
     bool _preferClaimedTokens,
     string calldata _memo,
     bytes calldata _delegateMetadata
-  ) external payable override {
+  ) external payable override nonReentrant {
+    // Verify that this terminal is a terminal of provided project ID
+    if (!directory.isTerminalOf(_projectId, this)) {
```

Comment:
you can abstract this to an `requireTerminalMatch` modifier and reuse in addToBalance, whatchu think?

Reply:
sure that's DRYer